src/main/java/seedu/address/model/pet/Pet.java

```diff
@@ -31,6 +33,7 @@ public class Pet {
      */
     public Pet(Name name, Species species, Breed breed, Age age, Sex sex, Set<Tag> modelTags) {
         requireAllNonNull(name, species, breed, age, sex);
+        this.uid = name.name + uid_counter++;
```

Comment:
Isn't this prone to issues and bugs when Pets are deleted?

Additionally when we reload the application, I believe this causes a conflict in uid_counter repeating 0? Do let me know if I'm wrong on this!

Reply:
You're right the counter I implemented would restart from 0 each time the app is opened. I'll use the java.util.UUID package instead and an overloaded Pet constructor to store the UID in the storage and load it from the storage whenever the app is opened.